upup/pkg/fi/cloudup/apply_cluster.go

```diff
@@ -1418,6 +1417,24 @@ func (n *nodeUpConfigBuilder) BuildConfig(ig *kops.InstanceGroup, apiserverAddit
 		config.ApiserverAdditionalIPs = apiserverAdditionalIPs
 	}
 
+	bootConfig.MasterInternalName = cluster.Spec.MasterInternalName
+	if bootConfig.MasterInternalName == "" {
+		for _, addr := range apiserverAdditionalIPs {
+			if net.ParseIP(addr).IsPrivate() {
+				bootConfig.MasterInternalName = addr
```

Comment:
Now I'm wondering if we should make `MasterInternalName` a []string, or even a list of objects, like the `/api` endpoint:

```
>  kubectl get --raw /api
{"kind":"APIVersions","versions":["v1"],"serverAddressByClientCIDRs":[{"clientCIDR":"0.0.0.0/0","serverAddress":"1.2.3.4:443"}]}
```

I guess because the version of nodeup is tightly coupled it doesn't much matter, this is easy to change going forward if we need to.